src/context/AuthStore.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { useReducer, useEffect } from 'react';
+import { AuthPayload, Mutation } from 'types';
+import { useMutation } from '@apollo/react-hooks';
+import { auth } from 'utils';
+import LOGIN_MUTATION from 'graphql/login.graphql';
+
+enum ActionTypes {
+  AUTH_INIT = 'auth/init',
+  AUTH_ERROR = 'auth/error',
+  AUTH_SUCCESS = 'auth/success',
+  AUTH_LOGOUT = 'auth/logout',
+}
+
+interface State {
+  loading: boolean;
+  error: any;
+  data: AuthPayload;
+  isAuthenticated: boolean;
+}
+
+interface Action {
+  type: string;
+  payload?: any;
+}
+
+export interface AuthContextData extends State {
+  login: (email: string, password: string) => Promise<any>;
+  logout: () => void;
+}
+
+const defaultState: State = {
+  loading: false,
+  error: null,
+  data: {} as AuthPayload,
+  isAuthenticated: false,
+};
+
+const defaultContext: AuthContextData = {
+  ...defaultState,
+  login: () => new Promise(() => null),
+  logout: () => null,
+};
+
+export const AuthContext = React.createContext(defaultContext);
+
+const authReducer = (state: State, action: Action) => {
```

Comment:
Maybe putting `authReducer` into separate file and import it will be good idea in terms of readability ?

Reply:
Ok, sure. 👍We can separate the things